### ISSUE TITLE
compile error "void value not ignored" (Ubuntu 16.04)

### ISSUE THREAD

**Issue by danielnaber** [nuspell/nuspell#25]
I've checked out the latest code, then called `./configure --with-boost=/prg/boost_1_67_0`, then `make`. It fails with this message:

```
/usr/include/c++/5/bits/stl_algo.h:566:32:   required from ‘_IIter std::find_if_not(_IIter, _IIter, _Predicate) [with _IIter = const wchar_t*; _Predicate = nuspell::my_ctype<wchar_t>::do_scan_not(std::ctype_base::mask, const char_type*, const char_type*) const::<lambda(auto:3)>]’
locale_utils.cxx:713:57:   required from here
/usr/include/c++/5/bits/predefined_ops.h:296:31: error: void value not ignored as it ought to be
  { return !bool(_M_pred(*__it)); }
                               ^
```

Any idea?

**Comment by dimztimz** [comment]
Use GCC 7 or later.

**Comment by dimztimz** [comment]
With the latest version, building with gcc 5 and thus building on Ubuntu 16.04 is supported. Newer boost is still required than the one shipped in ubuntu 16.04.
